packages/react-components/src/components/Switch/Switch.tsx

```diff
@@ -42,9 +49,41 @@ export const Switch: React.FC<SwitchProps> = ({
     }
   }, [on]);
 
-  const iconSize: IconSize = size === 'basic' ? 'small' : 'xsmall';
+  const getIcon = (iconSize: IconSize): JSX.Element => {
+    if (loading) {
+      return <Loader size="cover" />;
+    }
+
+    if (disabled) {
+      return (
+        <Icon
+          data-testid={`${baseClass}__disabled-icon`}
+          size={iconSize}
+          source={LockIcon}
+          kind={'primary'}
+        />
+      );
+    }
+
+    if (on) {
+      return (
+        <Icon
+          className={styles[`${baseClass}__icon--on`]}
```

Comment:
There is an `inconsistency` ->
`disabled-icon` vs `icon--on`

Reply:
Fixed it -> I've removed the incon--on (as we don't need it for additional styling of icon inside). Moreover I've renamed the data-testid to just `disabled-icon`